components/kubernetes/component.go

```diff
@@ -0,0 +1,99 @@
+package kubernetes
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/DataDog/test-infra-definitions/common/config"
+	"github.com/pulumi/pulumi/sdk/v3/go/auto"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+type Importable interface {
+	Key() string
+	Import(output auto.OutputMap) error
+}
+
+// The generic component resource that be created by any provider (kind, EKS, etc.)
+// We could apply the same thing for VM with AWS VMs, Azure VMs, and so on
+type KubernetesCluster struct {
+	pulumi.ResourceState
+	exportName string
+
+	KubeConfig pulumi.StringOutput `export:"kubeConfig"`
+}
+
+// The target type that's going to be deserialized on the other side.
+// We could unit test the matching if required.
+// The E stands for exported, probably can find something better
+type EKubernetesCluster struct {
+	Key string
+
+	KubeConfig string `json:"kubeConfig"`
+}
+
+// func NewEKubernetesCluster could be implemented with clusterName parameter to be able
+// to deserialize when Export is done inside predefined runFunc.
+// We could also implement a scan of OutputMap and the relevant key auto if only one if this kind present.
+
+// We can have a base class that implements Import and Key as there's no value in copy pasting
+func (ek *EKubernetesCluster) Import(output auto.OutputMap) error {
+	if ek.Key == "" {
+		return errors.New("missing key")
+	}
+
+	val, found := output[ek.Key]
+	if !found {
+		return errors.New("output not found")
+	}
+
+	// it *might* happen that val.Value is a map[interface{}]interface{} or something, which
+	// would mean that we need to implement our own reflection as well to import.
+	return json.Unmarshal(val.Value.([]byte), ek)
+}
+
+// Boilerplate component resource creation
+func newClusterComponent(e *config.CommonEnvironment, name string, opts ...pulumi.ResourceOption) (*KubernetesCluster, error) {
+	comp := &KubernetesCluster{
+		// as we components we can guarantee key unicity without asking user to provide name again
+		exportName: e.CommonNamer.ResourceName("dd", "kubeCluster", name),
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do look for something to sincerely praise. -->
**👏 praise**